apps/remote_control/lib/lexical/remote_control/code_intelligence/rename.ex

```diff
@@ -0,0 +1,164 @@
+defmodule Lexical.RemoteControl.CodeIntelligence.Rename do
+  alias Lexical.Ast
+  alias Lexical.Document
+  alias Lexical.Document.Edit
+  alias Lexical.Document.Line
+  alias Lexical.Document.Position
+  alias Lexical.RemoteControl.CodeIntelligence.Entity
+  alias Lexical.RemoteControl.Search.Store
+  require Logger
+
+  import Line
+
+  @spec rename(Document.t(), Position.t(), String.t()) ::
+          {:ok, %{Lexical.uri() => [Edit.t()]}} | {:error, term()}
+  def rename(%Document{} = document, %Position{} = position, new_name) do
+    with {:ok, entity, range} <- resolve(document, position),
+         {:ok, results} <- search(document, position, entity, range) do
+      to_uri_with_changes(results, new_name)
+    end
+  end
+
+  defp resolve(document, position) do
+    case Entity.resolve(document, position) do
+      {:ok, {:module, module}, range} -> {:ok, module, range}
+      {:error, error} -> {:error, error}
```

Comment:
While the other PR hasn't landed yet, I think this should be a catch-all so that it doesn't have to change (much) when it does. 

```suggestion
      _ -> {:error, :unsupported_entity}
```

Reply:
Also, just as a note/reminder, when the other PR lands, resolve will also potentially return `:struct`, which should be included for renaming. (It returns `:struct` for cursor positions like `%|MyStruct{}`.)